lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name] || info[:dmv_office] || info[:office_name] 
+    @address = info[:address] || info[:address_li] || info[:street_address_line_1] || info[:address1]
+    @phone = info[:phone] || info[:public_phone_number] 
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if !@services.include?('Vehicle Registration')
+      nil
+    else
+      vehicle.registration_date = Date.today.year
+      @registered_vehicles << vehicle
+      if vehicle.antique?
```

Comment:
Your logic is good here. Let's try to avoid nested if statements in the future. We can achieve this by using boolean logic operatos so your if statement on line 23 could be `if @services.include?("Vehicle Registration") && `vehicle.antique?`